51 add showConfirmationDialog to nav service

nickmeinhold opened the pull request:
Fixes #51

@tamari-gray hey mate I want to get on with other things for a bit but wanted to get this done so I've put it up - I haven't added any tests or even tested manually yet so happy to do that if you don't get to it first, hoping it'll just_work™️ 

not sure if this is a great solution, happy to talk it through any time if you want.  thanks! 

EDIT: and of course there's a failing test, definitely should have tried locally.  I had a quick hack and found a potential blocker, put up #64

nickmeinhold committed:
51 add showConfirmationDialog to nav service
Fixes #51

tamari-gray commented:
thanks heaps nick! No problem at all :) Hope you have a productive day!! Ill see you later 😄

nickmeinhold committed:
Merge branch 'dev' into nickmeinhold/issue51

nickmeinhold committed:
Fix tests

- added a Provider for NavigationService
- did a little cheeky renaming while I was there :-)
- added parameter to MyApp so NavigationService can be created in main

nickmeinhold commented:
So good that you have all these great tests @tamari-gray , there's no sneaking bad code through here!  

Ready for review whenever it suits... thanks!

tamari-gray reviewed (approved):
love it, everything looks great! 
thanks so much @nickmeinhold 🙏
